frontend/src/lib/components/PUA/changeDiscountPage.svelte

```diff
@@ -0,0 +1,164 @@
+<script lang="ts">
+	import close from '$lib/assets/close.svg';
+	import ChangeDiscountInput from './changeDiscountInput.svelte';
+	export let showModel = false;
+	export let changePageData: {
+		discount_name: string;
+		discount_description: string;
+		discount_max_price: number;
+		discount_id: number;
+		status: number;
+	};
+
+	export let discountData = {
+		kind: 'Shipping Discount',
+		how: 'NT$',
+		way: 'free shipping'
+	};
+
+	export let dis_haved: boolean;
+
+	let currentData: {
+		discount_name: string;
+		discount_description: string;
+		discount_max_price: number;
+		discount_id: number;
+		status: number;
+	};
+	currentData = { ...changePageData };
+
+	$: {
+		if (!showModel) {
+			initPage();
+		}
+		if (!dis_haved) {
+			initPage();
+		}
+	}
+	function initPage() {
+		dataError = { maxquantity_err: false, name_err: false, description_err: false };
+		currentData = { ...changePageData };
+	}
+	function toggleModel() {
+		showModel = !showModel;
+	}
+	let dataError: { maxquantity_err: boolean; name_err: boolean; description_err: boolean } = {
+		maxquantity_err: false,
+		name_err: false,
+		description_err: false
+	};
+
+	function saveCurrentData() {
+		if (
+			currentData.discount_max_price.toString() !== '' &&
+			currentData.discount_name !== '' &&
+			currentData.discount_description !== '' &&
+			/^\d+$/.test(currentData.discount_max_price.toString())
+		) {
+			changePageData = { ...currentData };
+			dis_haved = true;
+			dataError = { maxquantity_err: false, name_err: false, description_err: false };
+			toggleModel();
+		} else {
+			if (currentData.discount_max_price.toString() === '') {
+				dataError.maxquantity_err = true;
+			} else if (currentData.discount_max_price.toString() !== '') {
+				dataError.maxquantity_err = !/^\d+$/.test(currentData.discount_max_price.toString());
+			}
+			if (currentData.discount_name === '') {
+				dataError.name_err = true;
+			} else if (currentData.discount_name !== '') {
+				dataError.name_err = false;
+			}
+			if (currentData.discount_description === '') {
+				dataError.description_err = true;
+			} else if (currentData.discount_description !== '') {
+				dataError.description_err = false;
+			}
+		}
+	}
```

Comment:
use `<form>` and `<input>` with `required` for builtin input validation

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required

Reply:
also tailwind supports styling for it
https://tailwindcss.com/docs/hover-focus-and-other-states#form-states

Reply:
fixed